src/workerd/io/worker.c++

```diff
@@ -2707,9 +2705,13 @@ void Worker::Isolate::logWarning(kj::StringPtr description, Lock& lock) {
     });
   }
 
-  // TODO(now): should these be logged to stderr like `console.warn()`?
-  // Run with --verbose to log JS exceptions to stderr. Useful when running tests.
-  KJ_LOG(INFO, "console warning", description);
+  if (consoleMode == ConsoleMode::INSPECTOR_ONLY) {
+    // Run with --verbose to log JS exceptions to stderr. Useful when running tests.
+    KJ_LOG(INFO, "console warning", description);
+  } else {
+    fprintf(stderr, "%s\n", description.cStr());
+    fflush(stderr);
```

Comment:
stderr is unbuffered, `fflush` probably isn't helping here.

Reply:
Yeah I thought it was buffered too, that's surprising. An alternative might be to use `setvbuf()`/`setlinebuf()` to enable line buffering for stderr, but that would affect it outside of this scope too.